pyntrinio/pyntrinio.py

```diff
@@ -189,14 +189,54 @@ def gather_stock_time_series(api_key, ticker, start_date, end_date, output_forma
   Example
   -----------
   >>> gather_stock_time_series(api_key, 'AAPL')
-  {'ticker':['AAPL'], 'adj_close':[300], 'date':[datetime.date(2020, 1, 2)]}
   """
+  try:
+    # change dates to datetime objects
+    if start_date is not None:
+      start_date = datetime.strptime(start_date, '%Y-%m-%d').date()
+    if end_date is not None:
+      end_date = datetime.strptime(end_date, '%Y-%m-%d').date()
+  except:
```

Comment:
It's better to also check if `end_date` >= `start_date`. Because if we don't explicitly do this here, it would be checked as an `api_key` error later. 
```suggestion
    if start_date > end_date:
      print("Invalid Input: `end_date` is earlier than `start_date`.")
      return
  except:  
```

Reply:
Telling from the tests, it seems you've considered this, but strangely, I wasn't able to find the parts in the function file.

Reply:
Must have accidentally deleted this in the function - good catch! added back in